### ISSUE TITLE
`yarn install` not called in `assets:precompile` in production environment

### ISSUE THREAD

**Issue by yoonwaiyan** [rails/webpacker#1760]
In one of our project we added webpacker gem to a Rails app that was upgraded to Rails 5.1 from  version 4.x, which is having an issue while deploying to our production environment using capistrano that apparently caused by `yarn install` not being called while precompiling assets. After some research we are stumbled against this rake task which checks if it's Rails 5.1 to decide whether to run `yarn install` or not:

https://github.com/rails/webpacker/blob/001654fa3c92f9572438b2686a16f11e41cd246b/lib/tasks/webpacker/compile.rake#L11-L19

Is there anything we have missed up while upgrading from Rails 4 that could have call yarn install automatically while deploying to our production environment? We're using webpacker gem version of 3.5.5.

**Comment by Welziewagers2015** [comment]
Thanks

**Comment by larissa** [comment]
Do you have a `bin/yarn`? Older Rails versions didn't add a binstub for yarn, so unless you re-created binstubs after upgrading your Rails version, you're probably missing that file. The `yarn:install` rake task runs yarn using the binstub file, so this task will fail during runtime because `bin/yarn` doesn't exist, causing your node modules to not be installed.

**Comment by yoonwaiyan** [comment]
Hi @larissa,

I've re-populated the binstubs but it doesn't seem to work. This is the error I get in capistrano:

```
00:49 deploy:assets:precompile
      01 /usr/local/rvm/bin/rvm 2.4.2 do bundle exec rake assets:precompile
      01 Webpacker is installed 🎉 🍰
      01 Using /var/www/project/releases/20181030040357/config/webpacker.yml file for setting up webpack paths
      01 Compiling…
      01 Compilation failed:
      01 /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:11:in `exec': No such file or directory - /var…
      01  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:11:in `block in run'
      01  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:10:in `chdir'
      01  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:10:in `run'
      01  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/runner.rb:6:in `run'
      01  from ./bin/webpack:15:in `<main>'
      01
      01
(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as root@45.32.229.187: rake exit status: 1
rake stdout: Webpacker is installed 🎉 🍰
Using /var/www/project/releases/20181030040357/config/webpacker.yml file for setting up webpack paths
Compiling…
Compilation failed:
/var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:11:in `exec': No such file or directory - /var/www/project/releases/20181030040357/node_modules/.bin/webpack (Errno::ENOENT)
  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:11:in `block in run'
  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:10:in `chdir'
  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/webpack_runner.rb:10:in `run'
  from /var/www/project/shared/bundle/ruby/2.4.0/gems/webpacker-3.5.5/lib/webpacker/runner.rb:6:in `run'
  from ./bin/webpack:15:in `<main>'
rake stderr: Nothing written
```

**Comment by joeadcock** [comment]
@yoonwaiyan Were you able to get this resolved?

For what it is worth, I had the same problem.  It turned out that `yarn install` was failing due to the node version on the server (16.x was installed, but a recently added package required 18.x).

Joe

**Comment by yoonwaiyan** [comment]
@joeadcock It's not officially resolved but we did a workaround to call `yarn install` manually during deployment before assets precompile, which seems to work without any side effect for now. 

I do hope there's a good fix to avoid the same problem for other apps that does the same thing. It could be the case where we have left out some files or a gem that would call `yarn install` automatically, but I'm not giving this problem a priority for now. Hope someone with this similar problem could solve this though.

I did saw a pull request shortly after this issue being raised but I'm sure if it solves this problem. Anyhow I'm dropping this issue for now.
